src/channel/mpmc.rs

```diff
@@ -771,7 +804,38 @@ mod if_alloc {
             where
                 T: Send,
             {
-                generic_channel::<parking_lot::RawMutex, T>(0)
+                generic_channel::<parking_lot::RawMutex, T, FixedHeapBuf<T>>(0)
+            }
+
+            /// A [`GenericSender`] implementation backed by [`parking_lot`].
+            ///
+            /// Uses a `GrowingHeapBuf` whose capacity grows dynamically up to
+            /// the given limit. Refer to [`GrowingHeapBuf`] for more information.
+            ///
+            /// [`GrowingHeapBuf`]: ../../buffer/struct.GrowingHeapBuf.html
+            pub type GrowingSender<T> =
+                GenericSender<parking_lot::RawMutex, T, GrowingHeapBuf<T>>;
+            /// A [`GenericReceiver`] implementation backed by [`parking_lot`].
+            ///
+            /// Uses a `GrowingHeapBuf` whose capacity grows dynamically up to
+            /// the given limit. Refer to [`GrowingHeapBuf`] for more information.
+            ///
+            /// [`GrowingHeapBuf`]: ../../buffer/struct.GrowingHeapBuf.html
+            pub type GrowingReceiver<T> =
+                GenericReceiver<parking_lot::RawMutex, T, GrowingHeapBuf<T>>;
+
+            /// Creates a new growing channel.
+            ///
+            /// Refer to [`generic_channel`] for details.
+            pub fn growing_channel<T>(
```

Comment:
I think actually I am now really in favor of just converting the default `shared::channel` to using `GrowingHeapBuf` instead of adding yet another typedef. The default typedefs are mainly convenience feature for people who want "just a channel". And for those the growing buffer might be the best fit, since it simply mimics what the other frameworks are doing. 

If you agree I think we can avoid the other typedefs.

Reply:
I tend to agree.